types/src/blob.rs

```diff
@@ -1,60 +1,44 @@
-use celestia_proto::tendermint::types as tmtypes;
-use serde::{Deserialize, Deserializer, Serialize};
+use serde::{Deserialize, Serialize};
+use tendermint_proto::v0_34::types::Blob as RawBlob;
+use tendermint_proto::Protobuf;
 
+use crate::nmt::Namespace;
+use crate::{Bytes, Error};
+
+// NOTE: We do not the `serce(try_from)` pattern for this type
+// becase JSON representation needs to have `commitment` field but
+// Protobuf definition doesn't.
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Blob {
-    blob: tmtypes::Blob,
-    commitment: Vec<u8>,
+    pub namespace: Namespace,
+    pub data: Bytes,
+    pub share_version: u8,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub commitment: Option<Bytes>,
```

Comment:
is an `Option` there kind of a todo? I'm not sure `celestia-node` allows for an empty commitment

Reply:
will check